kornia/enhance/normalize.py

```diff
@@ -99,6 +99,14 @@ def normalize(
     if isinstance(std, float):
         std = torch.tensor([std] * shape[1], device=data.device, dtype=data.dtype)
 
+    if isinstance(mean, tuple):
```

Comment:
Hey @shijianjian .

What I think it would make sense is to review everywhere in the code and allow only to provide `mean` and `std` as `torch.Tensor` at this stage.

So the conversion from whatever accepted input to `torch.Tensor` is in every `instance constructor`.

Like this you can skip all these checks and conversions at every functional iteration, and keep the code more mantainable

Reply:
Actually, I am not sure, are these functions that the user should be able to use from their own code (are they public) or are they to only be used by the internal `kornia` objects (therefore private)?

If so, I think this strategy should be followed in general and code will be better unittested maybe?

Reply:
> Hey @shijianjian .
> 
> 
> 
> What I think it would make sense is to review everywhere in the code and allow only to provide `mean` and `std` as `torch.Tensor` at this stage.
> 
> 
> 
> So the conversion from whatever accepted input to `torch.Tensor` is in every `instance constructor`.
> 
> 
> 
> Like this you can skip all these checks and conversions at every functional iteration, and keep the code more mantainable 

Correct. That is exactly what I meant. You only think about tensors for functional calls, and provide some easy API with nn.Module.